nix/overlay.nix

```diff
@@ -7,27 +7,37 @@ let
   };
   extraPython3Packages = extraPackages.python3Packages;
   sources = import ../npins;
-in
-{
-  python3 = prev.lib.attrsets.recursiveUpdate prev.python3 { pkgs = extraPython3Packages; };
+in {
+  python3 = prev.lib.attrsets.recursiveUpdate prev.python3 {
+    pkgs = extraPython3Packages;
+  };
 
   # RFC 166
   nixfmt = (import sources.nixfmt).default;
 
   # go through the motions to make a flake-incompat project use the build
   # inputs we want
-  pre-commit-hooks = final.callPackage "${sources.pre-commit-hooks}/nix/run.nix" {
-    tools = import "${sources.pre-commit-hooks}/nix/call-tools.nix" final;
-    gitignore-nix-src = sources.gitignore;
-    isFlakes = false;
-  };
+  # currently throws an eval-error:
+  #  error: attribute 'lib' missing
```

Comment:
@RaitoBezarius ^

Reply:
It seems it's lacking of an `inherit lib;` ? cc @fricklerhandwerk who wrote this code.

Reply:
Cannot reproduce, even with with `nix-shell --pure`.

Reply:
Same on my end. Both `direnv reload` and `nix-shell --pure` work when I checkout this PR.

I suspect the issue is with the setting of Nix in the Github action. My nix version is:

```console
$ nix --version
nix (Nix) 2.17.1
```

Reply:
Works now. Fixing the rest.